### ISSUE TITLE
Allow two-way binding syntax for callbacks inside used elements

### ISSUE THREAD

**Issue by Enyium** [slint-ui/slint#6373]
Consider [this SlintPad demo](https://slintpad.com/?snippet=import+%7B+Button%2C+VerticalBox+%7D+from+%22std-widgets.slint%22%3B%0A%0Acomponent+Foo+%7B%0A++++in+property+%3Cbool%3E+has-lower%3A+true%3B%0A%0A++++in+property+%3Cstring%3E+upper-text+%3C%3D%3E+upper.text%3B%0A++++in+property+%3Cstring%3E+lower-text%3B%0A%0A++++callback+clicked-upper+%3C%3D%3E+upper.clicked%3B%0A++++callback+clicked-lower%3B%0A%0A++++VerticalBox+%7B%0A++++++++alignment%3A+start%3B%0A%0A++++++++upper+%3A%3D+Button+%7B%7D%0A%0A++++++++if+has-lower+%3A+Button+%7B%0A++++++++++++%2F%2F+This+works%3A%0A++++++++++++text+%3C%3D%3E+root.lower-text%3B%0A%0A++++++++++++%2F%2F+Why+shouldn%27t+this+also+work%2C+if+the+two-way-binding%0A++++++++++++%2F%2F+syntax+is+okay+for+callbacks+when+declaring+them%2C%0A++++++++++++%2F%2F+and+for+properties+inside+used+elements%3F%0A++++++++++++%2F%2Fclicked+%3C%3D%3E+root.clicked-lower%3B%0A%0A++++++++++++%2F%2F+You+currently+have+to+use+this+syntax.+This+would%0A++++++++++++%2F%2F+require+you+to+pass+the+whole+parameter+list%3A%0A++++++++++++clicked+%3D%3E+%7B+root.clicked-lower%28%29%3B+%7D%0A++++++++%7D%0A++++%7D%0A%7D%0A%0Aexport+component+Demo+%7B%0A++++Foo+%7B%0A++++++++upper-text%3A+%220%22%3B%0A++++++++lower-text%3A+%220%22%3B%0A%0A++++++++clicked-upper+%3D%3E+%7B%0A++++++++++++self.upper-text+%3D+self.upper-text.to-float%28%29+%2B+1%3B%0A++++++++%7D%0A%0A++++++++clicked-lower+%3D%3E+%7B%0A++++++++++++self.lower-text+%3D+self.lower-text.to-float%28%29+%2B+1%3B%0A++++++++%7D%0A++++%7D%0A%7D%0A&style=native) (see comments):

```slint
import { Button, VerticalBox } from "std-widgets.slint";

component Foo {
    in property <bool> has-lower: true;

    in property <string> upper-text <=> upper.text;
    in property <string> lower-text;

    callback clicked-upper <=> upper.clicked;
    callback clicked-lower;

    VerticalBox {
        alignment: start;

        upper := Button {}

        if has-lower : Button {
            // This works:
            text <=> root.lower-text;

            // Why shouldn't this also work, if the two-way-binding
            // syntax is okay for callbacks when declaring them,
            // and for properties inside used elements?
            //clicked <=> root.clicked-lower;

            // You currently have to use this syntax. This would
            // require you to pass the whole parameter list:
            clicked => { root.clicked-lower(); }
        }
    }
}

export component Demo {
    Foo {
        upper-text: "0";
        lower-text: "0";

        clicked-upper => {
            self.upper-text = self.upper-text.to-float() + 1;
        }

        clicked-lower => {
            self.lower-text = self.lower-text.to-float() + 1;
        }
    }
}
```

**Comment by ogoffart** [comment]
True, we could transform it into a simple call there, instead of making this an error https://github.com/slint-ui/slint/blob/b2016239d182278d5af5e2aecf62d70ff7659460/internal/compiler/object_tree.rs#L1675

**Comment by crai0** [comment]
I think this should use the `=>` syntax since it is not a two-way binding. For example, the binding `foo <=> bar` would lead to `foo` calling `bar` but not the other way around. This also needs to have type checking to make sure the call signatures of the two callbacks are compatible.

Therefore, I propose a separate callback forwarding syntax as implemented in dc7e1f1fc7a74c0c54bf7deb58dcadc8ebbd1354. @ogoffart Please let me know what you think. In case you'd want to adopt these changes, I'd finish up the LSP completion and documentation and create a PR.

**Comment by tronical** [comment]
We discussed this in a group (@nigel @ogoffart @hunger @FloVanGH and myself) and agree with @crai0 's [comment](https://github.com/slint-ui/slint/issues/6373#issuecomment-2555794670) that this looks like a two-way binding, but it doesn't behave like one, and just from that binding syntax you can't see that it's not the same. However #8401 addresses the goal of simplifying callback forwarding.
